src/EnergyPlus/DaylightingManager.cc

```diff
@@ -6706,20 +6706,21 @@ void DayltgInteriorIllum(EnergyPlusData &state,
                 ++count;
                 // need to map back to the original order of the "loop" to not change all the other data structures
                 loop = thisDaylightControl.MapShdOrdToLoopNum(count);
-
-                ICtrl = state.dataSurface->Surface(IWin).activeWindowShadingControl;
-                int IS = findWinShadingStatus(state, IWin);
-                if (state.dataSurface->Surface(IWin).HasShadeControl) {
-                    if (state.dataSurface->SurfWinShadingFlag(IWin) == WinShadingType::GlassConditionallyLightened &&
-                        state.dataSurface->WindowShadingControl(ICtrl).ShadingControlType == WindowShadingControlType::MeetDaylIlumSetp &&
-                        !previously_shaded(loop)) {
-                        DILLSW(igroup) += thisDaylightControl.IllumFromWinAtRefPt(loop, IS, 1);
-                        previously_shaded(loop) = true;
-                    } else {
-                        if (!previously_shaded(loop)) {
-                            DILLUN(igroup) += thisDaylightControl.IllumFromWinAtRefPt(loop, IS, 1);
+                if (loop > 0) {
```

Comment:
Seems like `loop` should be declared locally every time it's set to avoid misuse outside the main loops or carryover from one to the next?